open-craft/edx-simple-theme
# [BB-1335] High-level simple theme

This theme adds many SASS variables which can be used to customize the basic styles of an Open edX instance using the `simple-theme` role.

**Testing instructions**:
* Clone the source repository of this PR locally.
* Add `lms/static/sass/partials/lms/theme/_variables-v1.scsss` and `lms/static/sass/_lms-overrides.scss` using the templates in [this configuration branch](https://github.com/open-craft/configuration/tree/guruprasad/simple-theme-changes/playbooks/roles/simple_theme/templates/lms/static/sass).
* Define the values for the SASS variables to overridden in `lms/static/sass/common-variables.scss`.
* Enable the theme, compile the SASS and collect all the static assets.
* Visit LMS to check if the overrides specified by the variables in `common-variables.scss` are applied.
* Alternatively, visit each running appserver in [this instance](https://stage.console.opencraft.com/instance/3920/) for illustration of some customizations to mimic client themes.

**Author notes and concerns**:
* The diff in this PR may not reflect the changes properly as all the files were built from scratch and this PR is only used as comparison against the existing simple theme skeleton. So it is better to review each file fully instead of reviewing the changes.
* Since this requires the corresponding configuration changes (mentioned above) and Ocim changes, this PR shouldn't be merged after review and approval.
* Since it is not possible to add variables to customize every single aspect of the user interface in the beginning, I have added what looks like a minimum viable set which should help us reimplement most of the simple client themes.
* There are so many buttons in the UI and a lot of them are styled differently. Since it is not possible to achieve a consistent theming, I have defined variables for primary and secondary button styles and have applied them in most, if not all places.
* Whatever additional customizations are needed for each client theme can be exposed via additional variables during the porting if it is generic and useful for all the users of the theme or overridden through the `SIMPLETHEME_EXTRA_SASS` ansible variable.
* This might still be rought around the edges, particularly during the initial review passes.
